cache.go

```
@@ -147,33 +160,51 @@ func (t *TableCache) Disconnected() {
 
 // populate adds data to the cache and places an event on the channel
 func (t *TableCache) populate(tableUpdates TableUpdates) {
+	if t.orm == nil || t.dbModel == nil {
+		panic("TableCache without DatabaseModel cannot be populated")
+	}
 	t.cacheMutex.Lock()
 	defer t.cacheMutex.Unlock()
-	for table, updates := range tableUpdates.Updates {
+	for table := range t.dbModel.Types() {
+		updates, ok := tableUpdates.Updates[table]
+		if !ok {
+			continue
+		}
 		var tCache *RowCache
-		var ok bool
 		if tCache, ok = t.cache[table]; !ok {
 			t.cache[table] = newRowCache()
 			tCache = t.cache[table]
 		}
 		tCache.mutex.Lock()
 		for uuid, row := range updates.Rows {
 			if !reflect.DeepEqual(row.New, Row{}) {
+				newModel, err := t.createModel(table, &row.New, uuid)
+				if err != nil {
+					panic(err)
```

As opposed to panicking we could have an error channel as part of the cache.
We can send the error here, and in the `Client` have a goroutine that blocks until it gets an error, then shuts down gracefully.

Unless we just ignore errors here because they probably shouldn't ever happen. WDYT?

(We can always improve this in a follow up)

---

Yes, I agree. I think that is what we discussed in the other PR.
I think an error channel / event is expropriate. Maybe even expose it to user as part of `NotificationHandler` interface?

For now, errors could happen if the Model does not match the installed schema. Which is something that can easily happen because we don't have deep schema vs Model validation at creation time. I wanted to add it but this PR was getting huge.

If we want to improve all this in a follow up PR, we should at least print a error message on stderr for now.

---

I'll see if I can add better Model validation so we can at least minimize the chances of errors for now.

---

I've added schema validation at startup. Errors should not happen.
Should I leave the panic for now to help catch early bugs until we implement the error channel in a follow up PR?

---

Leaving it for now is fine